### ISSUE TITLE
Add policies

### ISSUE THREAD

**Issue by buraksekili** [absmach/magistrala#1443]
**FEATURE REQUEST**

1. Is there an open issue addressing this request? If it does, please add a "+1" reaction to the
   existing issue, otherwise proceed to step 2.

This issue is related to #1347 

2. Describe the feature you are requesting, as well as the possible use case(s) for it.

The `Auth` service manages access control of entities. However, the current service is not feasible to establish an advanced access control mechanism. Therefore, we need an authorization system to check whether Mainflux entities (e.g. `Users`) are authorized to do certain operations (e.g. `creating a group`) on the objects (e.g. `Groups`). 

There are two widely used policy-based authorization systems called RBAC and ABAC. Besides these approaches, Google introduced its own authorization system, called Zanzibar. Zanzibar does not use RBAC or ABAC, but it does take a similar `Access Control Lists (ACLs)` approach. 

First of all, Zanzibar has to be flexible to control access control of millions (maybe trillions?) Google's users. This flexibility comes with Simple Data Model and Configuration Language for policy relations.
> "ACLs are collections of object-user or object-object relations represented as *relation tuples*." [1]

So, relation tuples are the data type for the ACLs. Moreover: 

> this data type includes set-algebraic operators such as intersection and union for specifying potentially complex access control policies in terms of those user-object relations. [1]

For example, ACLs can refer to other ACLs to establish sophisticated relations (`Members of a <group_1>` are allowed to modify `<group_1>` where being a `members of a <group_1>` is another policy relation of itself). 

For more details of Google's Zanzibar, refer to [1].

<br/>

> **Ory Keto** is the first and only open-source implementation of "Zanzibar: Google's Consistent, Global Authorization System" [2]

All relation tuples can be represented in JSON format. The example relation tuples can be found via https://github.com/ory/keto/tree/master/contrib/cat-videos-example/relation-tuples.

ORY Keto provides REST API and Protocol Buffers API  for interacting with the Keto. 
> The APIs are separated based on privileges into a read and write endpoint. 
...
All APIs are available to gRPC and REST clients, although feature parity is not always given. [3]

The Keto APIs consist of two main parts; `Read` and `Write`. `Read` APIs allow us to `check`, `list`, and `expand` the relation tuples. On the other hand, `Write` APIs allow us to change (e.g. creating or deleting) relation tuples. For a more detailed overview, please refer to [3].

To sum up; Keto is the only open-source alternative for Google's Zanzibar at the moment. Although it is not fully enriched Zanzibar's features, it provides adequate APIs for managing and checking relations ("permissions") with HTTP and gRPC [2].

**References**

[1] https://www.usenix.org/conference/atc19/presentation/pang
[2] https://www.ory.sh/keto/docs/
[3] https://www.ory.sh/keto/docs/concepts/api-overview/

3. Indicate the importance of this feature to you (must-have, should-have, nice-to-have).

must-have
